enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,244 @@
+
+Advantage of using containerd:
+* Containerd is widely adopted and supported by the open-source community.
+* With minimal effort, we get new features and bug fixes from the community and enterprise.
+* Better CPU performance and memory usage compared with Docker
+* Microsoft is a major contributor in Windows containerd development and moved to containerd as the default
+runtime for their Kubernetes offerings.
+* Most of the Windows-supported Kubernetes orchestrators already moved to containerd.
+
+Disadvantage of using other runtimes:
+* Mirantis supported dockershim will continue to use Docker. Using Docker as runtime has an impact on memory
+and CPU performance.
+* From the business point of view subscribing to Mirantis for dockershim support is not an ideal situation as Mirantis
+is our key competitor.
+* The amount of time and engineering efforts involves in making CRI-O to work for Windows is huge. This doesn't
+do business justification due to the high cost and time.
+* Less traction from the community to develop CRI-O for Windows.
+
+## Design Details
+
+We plan to ship containerd 1.6.0 to with WMCO. Containerd will be built as a submodule and bundled into the WMCO package.
+As part of the WMCO Windows instance workflow, Containerd service will be started before kubelet starts as kubelet
+must require containerd to be running. Once containerd is started as Windows service, the remaining services starting steps remain
+the same. Kubelet's dependency on Docker runtime will be removed and replaced by containerd. A separate folder will be
+created under C:\k\ to store containerd config and related files. In the BYOH case, even after containerd becomes the default
+runtime Docker will continue to run. Users can choose to uninstall Docker or use it for non-Kubernetes related requirements.
+As Docker service is handled by the user, it is up to the user to decide if Docker should continue to run or not.
+When it comes to MachineSet, the user can choose a VM image that doesn't have Docker. Containerd will be the only runtime.
+If the MachineSet uses a VM image that comes with Docker, containerd will run along with Docker service. But Kubernetes will
+use only containerd. Customers can stop the Docker service if it is no longer needed. Even if Docker service gets started at any time,
+it can co-exist with containerd.
+
+Steps to install containerd as service:
+- scp containerd/related executables into Windows VM
+- copy the files into `C:\k\bin` folder
+- create containerd config file
+- register containerd as service
+- start containerd as a Windows service
+
+## Network changes
+Current CNI/IPAM will be used for containerd and no change will be made in HNS-Network and HNS-Endpoint creation
+steps. The config file which will be used by containerd will point to the same CNI/IPAM executables.
+
+## WMCO CLI option
+WMCO CLI option 'runtime' will be used to make containerd as the default runtime.  In the 4.10 community operator, it
```

Comment:
`WMCO CLI option 'runtime' will be used to choose containerd as the runtime during the OpenShift 4.10 / WMCO 5.0.0 development cycle`.